symengine/visitor.cpp

```diff
@@ -134,6 +134,53 @@ set_basic free_symbols(const Basic &b)
     return visitor.apply(b);
 }
 
+class FunctionSymbolsVisitor : public BaseVisitor<FunctionSymbolsVisitor>
+{
+public:
+    set_basic s;
+    uset_basic v;
+
+    void bvisit(const Subs &x)
+    {
+        set_basic set_ = function_symbols(*x.get_arg());
+        for (const auto &p : x.get_variables()) {
+            set_.erase(p);
+        }
```

Comment:
FunctionSymbols are not in `x.get_variables()` so this whole method can be removed